model/service/rdf/RdfBatteryService.php

```diff
@@ -162,6 +162,28 @@ public function findDeliveryBattery(\core_kernel_classes_Resource $delivery)
         return $batteries;
     }
 
+    /**
+     * Find all batteries without any delivery
+     *
+     * @return array|RdfBattery[]
+     * @throws BatteryException
+     */
+    public function findEmptyBatteries()
+    {
+        $batteries = [];
+
+        /** @var \core_kernel_classes_Resource $instance */
+        foreach ($this->getClass(self::BATTERY_URI)->getInstances(true) as $instance) {
+            $battery = $this->buildBattery($instance->getUri());
+            if ($this->getBatteryDeliveries($battery)) {
```

Comment:
the `getBatteryDeliveries` method will provide valid deliveries. Do u want to delete battery with not valid deliveries?

Reply:
In this case, I skip all batteries which have at least one delivery (valid or not it doesn't matter)

Reply:
make sense

Reply:
ok, with empty http://www.taotesting.com/ontologies/battery.rdf#deliveries property

Reply:
If `#deliveries` properties are empty it means that battery is broken or empty so it is OK I guess. If `isValidDelivery` returns an empty array (because deliveries are not valid), the battery will be marked as empty and will be deleted.

Reply:
if `isValidDelivery` returns an empty array, it does not mean that there is no delivery. A delivery with a future time of execution will be ignored. But you have to keep this battery